parity/helpers.rs

```diff
@@ -298,6 +300,12 @@ pub fn password_prompt() -> Result<Password, String> {
 	Ok(password)
 }
 
+pub fn new_fetch_client(https_proxy: Option<&String>) -> Result<fetch::Client, String> {
+	return https_proxy
+		.map_or_else(|| fetch::Client::new(), |s| fetch::Client::new_with_proxy(Some(s.to_string())))
```

Comment:
Pretty sure that `Some(s.to_string())` is not good a practice, but could find a way to do this idiomatically.

Reply:
It is fine here because, in the end, you need to take ownership of the String in the ```background thread (closure)```.
However, you could pass to the `String` by reference instead and clone when you need it! A String in Rust is a ```Vec<u8>``` which takes 3 words on the stack and the rest on the heap!

However, remove needless `return` and the ending `;`